### DISCUSSION TITLE
If Streaming or Deferring Loader Data is not working, and you're using an Nginx proxy this might be of interest to you

### DISCUSSION THREAD

**Discussion by wouterds** [remix-run/remix#7965] [Show and tell / tips]
I learned the hard way that apparently deferred loading of data through streaming does not work if your app is running behind a very basic nginx proxy with default settings.

Instead of getting a quick response while the rest of the data was streamed I noticed I was getting hit with a very long TTFB.
![](https://user-images.githubusercontent.com/1210628/282061834-0e3f0c89-d10d-45ac-b188-600697080267.jpg)

After debugging a little and setting up a [simple reproducible example](https://github.com/wouterds/remix-streaming) I found out it could be fixed by setting `proxy_buffering off` in my nginx config (https://github.com/wouterds/remix-streaming/commit/bd2fa6df5b8b82f44969f9ad21a260e335ec83da). As the name implies this setting buffers the upstream response and waits until it is ready before responding.

This fixed the issue and now we can make use of Remix streaming behind an nginx proxy.
![](https://user-images.githubusercontent.com/1210628/282061773-0dcc22be-922b-4854-8a4d-14c058bb05f8.jpg)

**Comment by block2k** [comment]
cool 🧨

**Comment by Duong-GM** [comment]
do you notice that on the first time, even tho Remix does stream content normally, browser have to wait until the document request is finished before rendering?
Subsequent call seems fine.
@wouterds
